### PR TITLE
Update datadog to 5.7.4

### PR DESCRIPTION
**Pull request by thedebugger**
###### Things done
- [y] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [y] NixOS
  - [n] OS X
  - [n] Linux
- [y] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [y] Tested execution of all binary files (usually in `./result/bin/`)
- [y] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by FRidh** [comment]
Is there a good reason `buildPythonPackage` is not used? Because dd-agent seems installable with setuptools.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -26937,4 +26937,113 @@ in modules // {
...
+  uptime = buildPythonPackage rec {
+    name = "uptime-${version}";
+    version = "3.0.1";
+
+    src = pkgs.fetchFromGitHub {
+      owner = "Cairnarvon";
+      repo = "uptime";
+      rev = "1ddfd06bb300c00e6dc4bd2a9ddf9bf1aa27b1bb";
+      sha256 = "148rqj0asw65c3z4whnqphfmkhlm181p296b7pxc1x79n97gn1pf";
+    };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: please use fetchurl with an archive from PyPI
https://pypi.python.org/pypi/uptime/3.0.1

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -26937,4 +26937,113 @@ in modules // {
...
+  pytestHttpbin = buildPythonPackage rec {
+    name = "pytest-httpbin-${version}";
+    version = "0.2.0";
+
+    src = pkgs.fetchFromGitHub {
+      owner = "kevin1024";
+      repo = "pytest-httpbin";
+      rev = "ae314b2a18eecac2a9d3b2034bfede9805ce7907";
+      sha256 = "0j5rcqlh59sqcv6csnk0i73xi5p43h32ynnz550q8pv9q68lzqa2";
+    };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: same here https://pypi.python.org/pypi/pytest-httpbin/0.2.3
Or is the specific version required?
In that case https://pypi.python.org/pypi/pytest-httpbin/0.2.0

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -26937,4 +26937,113 @@ in modules // {
...
+  vcrpy = buildPythonPackage rec {
+    name = "vcrpy-${version}";
+    version = "1.7.4";
+
+    src = pkgs.fetchFromGitHub {
+      owner = "kevin1024";
+      repo = "vcrpy";
+      rev = "v${version}";
+      sha256 = "1inm22xlnlb4xxcrrmrm3j73kffksb2838ypgykjbsjid8xvqdsd";
+    };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: here as well https://pypi.python.org/pypi/vcrpy/1.7.4

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -26937,4 +26937,113 @@ in modules // {
...
+  pyvmomi = buildPythonPackage rec {
+    name = "pyvmomi-${version}";
+    version = "6.0.0";
+
+    src = pkgs.fetchFromGitHub {
+      owner = "vmware";
+      repo = "pyvmomi";
+      rev = "51a54c0ca3c07aa7c3d4314d962efbce2d5b81f4";
+      sha256 = "1qxrbrrfy82dl6yqg6mffwvyzv0xwxfqpp61p1mf7ngy0cxsac50";
+    };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: and https://pypi.python.org/pypi/pyvmomi/6.0.0
or 
https://pypi.python.org/pypi/pyvmomi/6.0.0.2016.4

**Comment by thedebugger** [comment]
@FRidh thanks for the feedback. i'll change the github repo to pyvi. Re: dd-agent mkDerivation, it was like that before; not sure why chose it over pythonPackage. I can try to refactor it and see if that works.

**Comment by FRidh** [comment]
@thedebugger if you like to, go ahead. I won't stop you :-) But it is not necessary.

**Comment by joachifm** [comment]
@thedebugger still working on this?

**Comment by thedebugger** [comment]
@joachifm i think we can merge it if it looks okay. I didn't get a chance to refactor dd-agent to use pythonPackages :(

**Comment by FRidh** [comment]
note that 5.8.5 is the latest version by now.

**Comment by FRidh** [comment]
Please reopen when it seems ready to be merged.

**Pull request closed without merging**